src/application/engine/entities/BoDCharacter.java

```diff
@@ -37,6 +37,11 @@
      * @param speed
      *            Movement speed of the character.
      */
+    public BoDCharacter(int id)
```

Comment:
every character must have a body, physics, view, health and so on. it would make more sense to require those in the constructor than setting them explicitly in each derived class